lib/Authentication.ts

```diff
@@ -49,6 +52,150 @@ export default class Authentication {
     this.factory = new CryptoFactory(options.cryptoSuites || [new RsaCryptoSuite()]);
   }
 
+  /**
+   * Signs the AuthenticationRequest with the private key of the Requester and returns the signed JWT.
+   * @param request well-formed AuthenticationRequest object
+   * @param responseDid DID of the requester.
+   */
+  public async signAuthenticationRequest (request: AuthenticationRequest): Promise<string> {
+    if (request.response_type !== 'id_token' || request.scope !== 'openid') {
+      throw new Error('Authentication Request not formed correctly');
+    }
+    const requesterDid = request.iss;
+    let key: PrivateKey | undefined;

```

Comment:
Use `const` for `key` buy merging with the following line.

Reply:
```suggestion
    const key = this.getKey(requesterDid);
```